src/librustc_trans/debuginfo/gdb.rs

```diff
@@ -22,19 +23,13 @@ use std::ptr;
 use syntax::attr;
 
 
-/// Inserts a side-effect free instruction sequence that makes sure that the
-/// .debug_gdb_scripts global is referenced, so it isn't removed by the linker.
-pub fn insert_reference_to_gdb_debug_scripts_section_global(ccx: &CrateContext, builder: &Builder) {
+/// Inserts the .debug_gdb_scripts global into the set of symbols
+/// to be placed in `llvm.used`, so it isn't removed by the linker.
```

Comment:
"by the linker" - you mean the compiler, right? The linker doesn't know or care about llvm.used stuff but likely it has some special case to preserve the .debug_gdb_scripts section.